### PR TITLE
added liveness and readiness probes to debug container partials template

### PR DESCRIPTION
**Pull request by deusxanima**
**Subject**

added liveness and readiness probes to debug container partials template

**Problem**

Some environments have security restrictions which require liveness and readiness probes in order to deploy any container in the cluster. Currently our debug container does not contain these, and there is no easy way for users to add the values into the container due to the way it's injected into the pod but not propagated to the deployment.

**Solution**

Added static liveness and readiness probes to debug container partials template which will always pass and evaluate to true.

Fixes #[[11307](https://github.com/linkerd/linkerd2/issues/11307)]

Alen Haric (deusxanima) aharic88@gmail.com

### CODE REVIEW

File: charts/linkerd-control-plane/values-ha.yaml [hunk 1]
```diff
@@ -53,5 +53,8 @@ webhookFailurePolicy: Fail
 # service profile validator configuration
 spValidatorResources: *controller_resources
 
+# policy controller configuration
+policyControllerResources: *controller_resources
```
Comment (olix0r) [inline charts/linkerd-control-plane/values-ha.yaml:57]: These changes are duped from #11306. We'll want to either remove these from the branch or merge that PR first and then merge main into this branch...

File: charts/linkerd-control-plane/values.yaml [hunk 2]
```diff
@@ -502,6 +502,9 @@ policyValidator:
   # for more information.
   injectCaFromSecret: ""
 
+# -|- CPU, Memory and Ephemeral Storage resources required by the policy controller 
+#policyControllerResources:
+
```
Comment (olix0r) [inline charts/linkerd-control-plane/values.yaml:507]:
```suggestion
```

File: charts/linkerd-control-plane/values-ha.yaml [hunk 3]
```diff
@@ -53,5 +53,8 @@ webhookFailurePolicy: Fail
...
+
```
Comment (olix0r) [inline charts/linkerd-control-plane/values-ha.yaml:58]:
```suggestion
```

**Review by olix0r — CHANGES_REQUESTED** [review]

**Review by alpeb — APPROVED** [review]
I've replaced the commands with just `true`, fixed the golden files and the pod-with-debug.patch.json file :pray:

**Review by adleong — APPROVED** [review]

**Review by mateiidavid — APPROVED** [review]

**Review by olix0r — APPROVED** [review]

**Pull request merged**